pkg/util/time_test.go

```diff
@@ -0,0 +1,174 @@
+package util
+
+import (
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func Test_SplitTimeRangeByResolution(t *testing.T) {
```

Comment:
Do you think it would make sense to add a few tests for ranges that span over 2 or more days? I know this should work as expected, but it might not be super obvious when looking at the tests only.

Reply:
Yep, added one